[llvm] install tools in tools/llvm

strega-nil opened the pull request:
This fixes #13294; however, I'm not sure this is the best solution, so I would like reviews from affected parties (this is also the first portfile I've ever written, haha!)

strega-nil committed:
[llvm] install tools in tools/llvm

strega-nil committed:
remove the giant commented out block of code

strega-nil committed:
update port-version

BillyONeal commented on ports/llvm/portfile.cmake:
```diff
@@ -145,57 +146,15 @@ vcpkg_install_cmake()
-        file(WRITE ${_target} "${_contents}")
-    endforeach()
+    file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/tools)
+    file(RENAME ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/tools/${PORT})
```
The `halide` failure is caused by

```
CMake Error at D:/installed/x64-windows/share/llvm/LLVMExports.cmake:716 (message):
  The imported target "LTO" references the file

     "D:/installed/x64-windows/debug/bin/LTOd.dll"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "D:/installed/x64-windows/share/llvm/LLVMExports.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  D:/installed/x64-windows/share/llvm/LLVMConfig.cmake:174 (include)
  C:/a/1/s/scripts/buildsystems/vcpkg.cmake:555 (_find_package)
  dependencies/llvm/CMakeLists.txt:12 (find_package)
```

I think this needs to use vcpkg_copy_tools or something like that instead so that only exes are moved?

BillyONeal commented on ports/llvm/portfile.cmake:
```diff
@@ -205,4 +164,3 @@ if("clang" IN_LIST FEATURES)
 endif()
 
 # Don't fail if the bin folder exists.
```
```suggestion
```

BillyONeal commented:
Thanks again for running this down!

strega-nil committed:
hopefully fix the port?

There are still some issues, namely that the python scripts and DLLs that are copied into `tools/llvm`
are not considered to be "installed by llvm", and thus are not removed when llvm is removed

strega-nil committed:
format!

strega-nil committed:
Merge remote-tracking branch 'upstream/master' into llvm

strega-nil committed:
apparently REGEX REPLACE fails if a thing doesn't match

strega-nil committed:
fix LLVM_REMOVE_EXTENSION_REGEX on windows

strega-nil committed:
actually read the cmake regex docs...

strega-nil committed:
fix the name of the variable

strega-nil committed:
turns out CMAKE_MATCH_1 is the one I want

strega-nil committed:
need to update VERSION for new policy

BillyONeal commented on toolsrc/src/vcpkg/base/files.cpp:
```diff
@@ -26,20 +26,6 @@
 
 namespace
 {
-#if defined(_WIN32)
```
Oops

strega-nil committed:
stop removing debug/bin

strega-nil committed:
Merge remote-tracking branch 'upstream/master' into llvm

strega-nil committed:
fix faulty merge
